### ISSUE TITLE
REE total-capacity clamp is non-differentiable at the constraint that binds

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#193]
## Problem

The total-capacity limiter added for #112 is a hard `jnp.where` threshold. It puts a kink in the gradient exactly at the capacity constraint, which is the constraint that binds at any economically optimal design.

## Evidence

`src/difflow_ree/units/extraction.py:264-286`

```python
# Enforce total extractant loading capacity (Bug #112)
if self._isotherm is not None:
    total_newly_extracted = sum(...)
    max_capacity = self._isotherm.max_ree_conc * F_org
    scale = jnp.where(
        total_newly_extracted > max_capacity,
        max_capacity / jnp.maximum(total_newly_extracted, 1e-10),
        1.0,
    )
```

`scale` is continuous at the switch point but its derivative is not: it is identically 0 below and `-max_capacity/total**2` above.

## Why it matters

This is a differentiable-flowsheet package, and the whole point is that `jax.grad` through a flowsheet is trustworthy. A gradient-based optimizer driving solvent flow down to reduce cost walks straight into this boundary, because operating at capacity is exactly what minimizes solvent inventory. At the kink the gradient reported on either side disagrees, so line searches stall and the optimizer chatters at its own optimum.

It is also silent: nothing in the returned `info` records that the clamp fired, so a user cannot tell a converged design from one pinned against a non-physical wall.

The fix for #112 was correct in identifying that the Kremser equation can predict extraction beyond physical capacity. The problem is that clamping the symptom is the wrong remedy in a differentiable model.

## Suggested fix

In order of preference:

1. **Remove the need for it.** If the stage solves an extractant balance (see #196), capacity is enforced by the balance itself and is smooth everywhere, because free extractant approaches zero continuously rather than being clipped. This is the real answer.

2. **If a limiter must remain in the correlation path**, make it smooth. A softmin over the two branches,

   ```python
   scale = max_capacity / (max_capacity + total_newly_extracted)   # or a logsumexp blend
   ```

   or `jax.nn.sigmoid`-weighted blending over a stated width, with the width exposed as a parameter so the user can tighten it as the solve converges.

3. **At minimum**, set `info["capacity_clamped"] = ...` and the clamped fraction, so the condition is observable.

Related: the same argument applies to `third_phase_loading_limit` at `extraction.py:502-507`, which sets an `info` flag only. In a design model, third-phase onset is an operating boundary the optimizer will cross because crossing it is profitable in the model. It should be available as an inequality constraint, not only a diagnostic.

## Test to add

`jax.test_util.check_grads` on recovery with respect to solvent flow, evaluated at a point where `total_newly_extracted == max_capacity`. Today this fails.

**Comment by jkitchin** [comment]
Fixed in #204, merged as `2862b04`.

The hard clamp is replaced by a smooth `(1 + (total/cap)^k)^(-1/k)` saturation with `k = capacity_sharpness` (default 8, reachable from the flowsheet Params), and the condition is reported through `info[]` — `theta_total`, `capacity_scale`, clamped fraction — so a converged design is distinguishable from one pinned at the wall. `cap` is `F_extractant/m`, a molar flow, rather than the previous concentration-times-flow.

Review found a second hard kink that this issue's diagnosis also covers: `jnp.maximum(1 - theta_solvent, 0)` on the entering-solvent path, with an identically-zero gradient beyond it — the "dead lever" `difflow.planning.health` warns about. The new capacity basis made it bite ~6x earlier than the code it replaced, on the path a recycled extract-strip solvent travels. It is smoothed the same way. Third-phase onset now also exposes a signed margin usable as an inequality constraint.

(This issue stayed open because the PR body wrote `Fixes #189, #190, ...` — GitHub needs a closing keyword before each number, so only #189 was parsed. Closing manually.)
